packages/core/src/pipeline/config/validation/imageProviderBeforeType.validator.ts

```diff
@@ -0,0 +1,30 @@
+import { IStageOrTriggerValidator, IValidatorConfig, PipelineConfigValidator } from './PipelineConfigValidator';
+import { ICluster, IPipeline, IStage, IStageOrTriggerTypeConfig } from '../../../domain';
+
+export interface IImageProviderBeforeTypeValidationConfig extends IValidatorConfig {
+  triggerTypes?: string[];
+  message: string;
+}
+
+export class ImageProviderBeforeTypeValidator implements IStageOrTriggerValidator {
```

Comment:
Worth documenting the purpose of this validator (the text in the PR for example)